### PR TITLE
Fix User-Agent header in requests made by Helm 

### PR DESCRIPTION
**Pull request by programmer04**
> This is the recreation of PR #10207 which has been closed due to problems kicking off CI for it.

Current Helm v3.8.1 installed from package manager or build from main branch when makes requests to K8s API (with use of K8s SDK), e.g. executing the command

```
helm history test
```
does not have the proper value set in `User-Agent` header. It looks like that on macOS Monterey with Apple Silicon.
```yaml
User-Agent: Go-http-client/1.1
```
This PR fixes it and makes it consistent with `User-Agent` set when requests are made to chart repositories (it works as expected). So now it always looks like that
```yaml
User-Agent: Helm/3.8+unreleased
```
or
```yaml
User-Agent: Helm/3.8
```

Hence I removed those comments too from the code
```
// TODO add user-agent
```
because it's been already done and works as expected for interaction with chart repos and etc. Here is the line which provides it https://github.com/helm/helm/blob/0d9ebc7885e427373fbdaa95f461b349fb0b1ac1/pkg/getter/httpgetter.go#L53

### CODE REVIEW

File: pkg/cli/environment_test.go
Comment (phil9909) [inline pkg/cli/environment_test.go:236]:
```suggestion
func TestUserAgentHeaderInK8sRESTClientConfig(t *testing.T) {
```

**Review by phil9909 — CHANGES_REQUESTED** [review]

**Review by davidkarlsen — APPROVED** [review]
Lgtm

**Review by phil9909 — APPROVED** [review]

File: pkg/cli/environment.go
Comment (mattfarina) [inline pkg/cli/environment.go:237]: My concern is that a `WrapConfigFn` is already setup. Won't this overwrite that? Should we combine them? See...

https://github.com/helm/helm/blob/5a8e91a010a923c3505c19954cfda4f01432bbc9/pkg/cli/environment.go#L118-L124

Comment (programmer04) [reply pkg/cli/environment.go:237]: Yeah, the lines mentioned by you were introduced after my original PR. Good catch! I am moving my change here

**Review by mattfarina — APPROVED** [review]
Note, this needs a second maintainer to approve before it can be merged.

**Comment by mattfarina** [comment]
@programmer04 could you please rebase this PR from main.

**Comment by programmer04** [comment]
Thanks for reaching me, it's done @mattfarina

**Review by hickeyma — APPROVED** [review]
LGTM, thanks @programmer04 for pushing new PR to fix CI issue.

**Pull request merged**
